core/math/vector2.h

```diff
@@ -114,6 +114,34 @@ struct _NO_DISCARD_ Vector2 {
 	_FORCE_INLINE_ Vector2 bezier_interpolate(const Vector2 &p_control_1, const Vector2 &p_control_2, const Vector2 &p_end, const real_t p_t) const;
 	_FORCE_INLINE_ Vector2 bezier_derivative(const Vector2 &p_control_1, const Vector2 &p_control_2, const Vector2 &p_end, const real_t p_t) const;
 
+	_FORCE_INLINE_ Vector2 rotate_toward(Vector2 p_to, real_t p_rotation_delta, real_t p_length_delta = 0.0f) {
```

Comment:
The method does not need to be as complicated as you have made it. It should be this simple:

* Calculate the `angle_to` between the vectors.
* If the angle is less than or equal to `p_rotation_delta`, return `p_to`.
* Else, if the angle is greater than `p_rotation_delta`, `slerp` by a factor of `p_rotation_delta / angle`.

The heavy lifting, including support for vectors of different lengths, is already handled by `slerp`.

Reply:
I agree that it did get a bit complicated, mainly because it's kind of two functions merged into one.
The idea behind it was that it handles the rotation on two normalized vectors, but still tries to move the length of `from` toward `to` using the second delta parameter. Main motivation being how it's currently done in Unity... If this behavior is not needed/wanted you should be able to just remove everything related to `p_length_delta`. It's easy enough to recreate this behavior by writing a custom function in engine that utilizes the base `rotate_toward` so I would not mind it.

Reply:
What is the expected use case of `p_length_delta`? I would not expect Vector2/3 `rotate_toward` to behave differently from angle/Basis/Quat/etc `rotate_toward` or differently from Vector2/3 `slerp`, so I would not add that level of complication unless it can be strongly justified.

Reply:
True, I cannot really think of any real world use cases of this behavior, sometimes I just take Unity as some kind of gold standard which it obviously isn't.

Reply:
Actually, thinking through it again, I did think of a semi-related use case: rotating a vector to another while keeping it's length, which would essentially be performing `rotate_toward` on two normalized vectors, then multiplying it by the original length of the first one. If this is important enough it could be added as a `normalized_rotation` bool at the end of the method.